### PR TITLE
Allow add items to array config via ENV

### PR DESCRIPTION
**Pull request by miry**
Allow to create a new templates or tags configs, if there are no records
in the default config. 

Support next cases:
- Without config file:

```
$ INFLUXDB_GRAPHITE_ENABLED=true INFLUXDB_GRAPHITE_TEMPLATES="stats.* .host.measurement.cpu,system.* .host.region.measurement*" influxd config

...
[[graphite]]
  enabled = true
  bind-address = ":2003"
  database = "graphite"
  retention-policy = ""
  protocol = "tcp"
  batch-size = 5000
  batch-pending = 10
  batch-timeout = "1s"
  consistency-level = "one"
  templates = ["stats.* .host.measurement.cpu", "system.* .host.region.measurement*"]
  separator = "."
  udp-read-buffer = 0

....
```
- With specified template rule:

``` toml
[[graphite]]
  enabled = true
  templates = ["stats-config.* .host.measurement.cpu"]
```

``` bash
$ INFLUXDB_GRAPHITE_ENABLED=true INFLUXDB_GRAPHITE_0_TEMPLATES_0="stats-over.* .host.measurement.cpu" influxd config
...
[[graphite]]
  enabled = true
  bind-address = ":2003"
  database = "graphite"
  retention-policy = ""
  protocol = "tcp"
  batch-size = 5000
  batch-pending = 10
  batch-timeout = "1s"
  consistency-level = "one"
  templates = ["stats-over.* .host.measurement.cpu"]
  separator = "."
  udp-read-buffer = 0

....
```

As you see from the first example `INFLUXDB_GRAPHITE_TEMPLATES` separate rules by `,`.

Fixes: https://github.com/influxdata/influxdb/issues/6943
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
###### Required only if applicable

_You can erase any checkboxes below this note if they are not applicable to your Pull Request._
- [x] Provide example syntax

### CODE REVIEW

**Comment by miry** [comment]
@gunnaraasen and @e-dard can you check this PR?

**Comment by e-dard** [comment]
@jackzampolin does this look like what you were after in #6943 ?

**Comment by jackzampolin** [comment]
@e-dard Yeah that's basically it. Want to make it easy to do graphite parsing by setting templates with ENV. I would also ask @gunnaraasen as he worked on this code too.

File: cmd/influxd/run/config.go [hunk 1]
```diff
@@ -172,117 +172,119 @@ func (c *Config) Validate() error {
...
+		if err != nil {
+			return fmt.Errorf("failed to apply %v to %v using type %v and value '%v'", prefix, structKey, element.Type().String(), value)
 
-			// Use the upper-case prefix and toml name for the env var
-			key := strings.ToUpper(configName)
-			if prefix != "" {
-				key = strings.ToUpper(fmt.Sprintf("%s_%s", prefix, configName))
+		}
+		element.SetFloat(floatValue)
+	case reflect.Slice:
```
Comment (gunnaraasen) [inline cmd/influxd/run/config.go]: This case should also handle `reflect.Array`. I don't think arrays are used in the configuration at this moment, but might as well cover all the bases.

Comment (miry) [inline cmd/influxd/run/config.go]: We have array as I understand correctly in next example there is `templates` option:

```
[[graphite]]
  protocol = "udp"
  templates = [
    "default.* .template.in.config"
  ]
```

File: cmd/influxd/run/config.go [hunk 2]
```diff
@@ -172,117 +172,119 @@ func (c *Config) Validate() error {
...
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		intValue, err := strconv.ParseUint(value, 0, element.Type().Bits())
+		if err != nil {
+			return fmt.Errorf("failed to apply %v to %v using type %v and value '%v'", prefix, structKey, element.Type().String(), value)
+		}
+		element.SetUint(intValue)
+	case reflect.Bool:
+		boolValue, err := strconv.ParseBool(value)
+		if err != nil {
+			return fmt.Errorf("failed to apply %v to %v using type %v and value '%v'", prefix, structKey, element.Type().String(), value)
```
Comment (miry) [inline cmd/influxd/run/config.go]: fixed. thank you.

File: cmd/influxd/run/config.go [hunk 3]
```diff
@@ -172,117 +172,119 @@ func (c *Config) Validate() error {
...
 
-			// Use the upper-case prefix and toml name for the env var
-			key := strings.ToUpper(configName)
-			if prefix != "" {
-				key = strings.ToUpper(fmt.Sprintf("%s_%s", prefix, configName))
+		}
+		element.SetFloat(floatValue)
+	case reflect.Slice:
+		for j := 0; j < element.Len(); j++ {
```
Comment (miry) [inline cmd/influxd/run/config.go]: Added comment. also added comment bellow few lines.

**Comment by miry** [comment]
@gunnaraasen I have updated code according the comments.

**Comment by e-dard** [comment]
@miry looks good. Can you add an entry to the `CHANGELOG` and make sure you give yourself credit!

**Comment by e-dard** [comment]
@miry can you also rebase from master and push. I think the test failure is because Circle is testing the head ref rather than the merge result of this PR and `master`. Not sure why it's doing that. The test failure looks completely unrelated to your changes, and when I pulled down your branch it's pretty far behind `master`.

**Comment by miry** [comment]
@e-dard sure

**Comment by miry** [comment]
@e-dard updated Changelog and rebased against master.

**Comment by fdrouet** [comment]
awesome fix !! when could we expect to see that in a released version of Influxdb ?

**Comment by burdandrei** [comment]
Great fix! waiting it to be merged!

**Comment by miry** [comment]
@e-dard @gunnaraasen @jackzampolin I have rebased against master.

**Comment by miry** [comment]
@e-dard who can merge this pr?

**Comment by jwilder** [comment]
Thanks @miry!

**Pull request merged**